### PR TITLE
Report error when ATEN_THEADING is OMP and USE_OPENMP is turned off.

### PR DESCRIPTION
**Pull request by xuhdev**
Currently, even if USE_OPENMP is turned off, ATEN_THEADING can still use OpenMP. This commit fixes it.

### CODE REVIEW

**Review by pbelevich — COMMENTED** [review]
can you please rebase?

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -22,12 +22,15 @@ set(AT_PARALLEL_NATIVE_TBB 0)
...
+  if(NOT USE_OPENMP)
+    message(FATAL_ERROR "ATen is using OpenMP backend but USE_OPENMP is off. Please either change ATEN_THREADING or turn on USE_OPENMP.")
+  endif()
   set(AT_PARALLEL_OPENMP 1)
 elseif("${ATEN_THREADING}" STREQUAL "NATIVE")
   set(AT_PARALLEL_NATIVE 1)
 elseif("${ATEN_THREADING}" STREQUAL "TBB")
   if(NOT USE_TBB)
-    message(FATAL_ERROR "Using TBB backend but USE_TBB is off")
+    message(FATAL_ERROR "ATen is using OpenMP backend but USE_TBB is off. Please either change ATEN_THREADING or turn on USE_TBB.")
```
Comment (peterjc123) [inline caffe2/CMakeLists.txt:33]: nit: Should be TBB backend, right?

Comment (xuhdev) [reply caffe2/CMakeLists.txt]: Right, fixed

**Review by peterjc123 — APPROVED** [review]

**Review by pbelevich — APPROVED** [review]

**Comment by pbelevich** [comment]
@xuhdev I reverted it, because people complained that MacOS builds stopped working with the error message from this diff, I tested it myself and I confirm it:
```
CMake Warning at cmake/Dependencies.cmake:1022 (message):
  Not compiling with OpenMP.  Suppress this warning with -DUSE_OPENMP=OFF
...
-- Using ATen parallel backend: OMP
CMake Error at caffe2/CMakeLists.txt:26 (message):
  ATen is using OpenMP backend but USE_OPENMP is off.  Please either change
  ATEN_THREADING or turn on USE_OPENMP.
```

**Comment by xuhdev** [comment]
Would the error be legitimate If ATen is using OpenMP and user intends to turn OpenMP off?

**Pull request closed without merging**